store/sqlstore/post_store.go

```diff
@@ -554,7 +554,7 @@ func (s *SqlPostStore) GetPostsSince(channelId string, time int64, allowFromCach
 
 	list := model.NewPostList()
 
-	var latestUpdate int64 = 0
+	latestUpdate := time
```

Comment:
So if I'm reading this correctly, if the `GetPostsSince()` call returns no messages, the time specified in the `since` parameter will be saved in the `lastPostTime` cache, which is also used as the posts etag for that channel.

This means that if the `since` parameter is *after* the time of the last post in the channel, it could blow away the cache for that channel incorrectly, having a potentially negative performance impact.

But then again, is the `since` value in normal usage always the time of the last known post, in which case the value in the cache either a) gets overwritten with the same value that's already in there, or b) get added to the cache because there wasn't currently a cached value for that channel. This would actual be (theoretically at least) a performance improvement in terms of more etag hits. :man_shrugging:

Reply:
That's a good point. It's not always the time of the last post in the channel since I think there's some cases where it's the last time that the websocket disconnected, so that may negatively affect the etag hit rate.

I'd lean towards leaving this as is for now since there shouldn't be any performance change since 5.12. We may want to separate the caches used by these two endpoints, but since there was some discussion of redoing the caching logic over the next few months, I don't think we need to change this now.

Reply:
OK. I'll file a ticket to the platform backlog to look into this with reference back to this ticket.